doc/RichTextArea/RichTextArea.md

```diff
@@ -0,0 +1,742 @@
+- TextPos **replace**(StyleResolver, TextPos start, TextPos end, StyledInput in, boolean allowUndo)
+
+The I/O interfaces **StyledInput** and **StyledOutput** provide the transport mechanism based on streaming of individual **StyledSegment**s.
+
+
+
+#### Clipboard
+
+The StyledTextModel provides capability to copy to and paste from the system clipboard in a variety of formats.  Several implementations are provided, all extending the **DataFormatHandler** class:
+
+|Format                             |Implementation                 |Export |Import |
+|:----------------------------------|:------------------------------|:------|:------|
+|DataFormat.HTML                    |**HtmlExportFormatHandler**    |yes    |no
+|DataFormat.PLAIN_TEXT              |**PlainTextFormatHandler**     |yes    |yes
+|RichTextFormatHandler.DATA_FORMAT  |**RichTextFormatHandler**      |yes    |yes
+|DataFormat.RTF                     |**RtfFormatHandler**           |yes    |yes (limited)
+
+The default model, EditableRichTextModel, copies plain text, HTML, RTF, as well as an internal format.  It also supports pasting plain text, internally formatted data, and a limited support for RTF.
+
+The following methods in StyledTextModel allow to register, remove, or query the data formats supported by the model:
+
+- Set<StyleAttribute<?>> **getSupportedAttributes**()
+- void **registerDataFormatHandler**(DataFormatHandler, boolean forExport, boolean forImport, int priority)
+- void **removeDataFormatHandler**(DataFormat, boolean forExport)
+  
+At the control level, the following methods allow for data transfer in any format supported by the model:
+
+- void **read**(DataFormat)
+- void **read**(DataFormat, InputStream)
+- void **write**(OutputStream)
+- void **write**(DataFormat, OutputStream)
+
+
+#### Adding New Attributes
+
+At the core of converting the style attributes in the model to the visual representation is a **StyleHandlerRegistry**.  This registry contains code (see **StyleAttributeHandler** interface) which applies the inline styles on either Text node for the text segment attributes (or TextFlow node for the paragraph attributes) during the process of rendering of document content by the skin.
+
+Because this process is linked to a skin, the registry is contained in the control.  Adding support for a new attribute therefore requires extending RichTextArea:
+
+1. declare the new StyledAttribute
+2. create a StyleAttributeHandler which will provide the inline style(s)
+3. extend RichTextArea and initialize the new StyleHandlerRegistry instance for the class, combining the parent class registry and the new handler(s)
+
+Example, based on the CodeArea class:
+
+```java
+public class CodeArea extends RichTextArea {
+    static final StyleAttribute<Font> FONT = new StyleAttribute<>("CodeArea.FONT", Font.class, false);
+    static final StyleAttribute<Integer> TAB_SIZE = new StyleAttribute<>("CodeArea.TAB_SIZE", Integer.class, true);
```

Comment:
How would an application use this newly added functionality?

Reply:
re-phrased